kvs/client/python/anna/ipc_client.py

```diff
@@ -12,59 +12,167 @@
 #  See the License for the specific language governing permissions and
 #  limitations under the License.
 
-GET_ADDR = "ipc:///requests/get"
-PUT_ADDR = "ipc:///requests/put"
+GET_REQUEST_ADDR = "ipc:///requests/get"
+PUT_REQUEST_ADDR = "ipc:///requests/put"
 
+GET_RESPONSE_ADDR_TEMPLATE = "ipc:///responses_%d/get"
+PUT_RESPONSE_ADDR_TEMPLATE = "ipc:///responses_%d/put"
+
+TIMEOUT = 5000
+
+from .functions_pb2 import *
 from .kvs_pb2 import *
 from .lattices import *
 import zmq
 
 class IpcAnnaClient:
-    def __init__(self, offset=0):
+    def __init__(self, thread_id = 0, offset=0):
         self.context = zmq.Context(1)
 
-        self.get_socket = self.context.socket(zmq.REQ)
-        self.get_socket.connect(GET_ADDR)
+        self.get_push_socket = self.context.socket(zmq.PUSH)
+        self.get_push_socket.connect(GET_REQUEST_ADDR)
+
+        self.put_push_socket = self.context.socket(zmq.PUSH)
+        self.put_push_socket.connect(PUT_REQUEST_ADDR)
+
+        self.get_pull_socket = self.context.socket(zmq.PULL)
```

Comment:
Let's call these `get_response_socket` and `pull_response_socket`.

Reply:
And call the `push` sockets `request` sockets instead.

Reply:
I was trying to match the name with socket type...
But yeah response and request read better.